src/data_analysis.py

```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+
+
+df = pd.read_csv('data/task3_dateset.csv', dtype=str)
+df['Date'] = pd.to_datetime(df['Date'], format='%m/%d/%y')
+df['Ads_Run'] = df['Ads_Run'].fillna(0)
+df['Ads_Run'] = df['Ads_Run'].astype(int)
+df = df.sort_values(by=['Site', 'Date'])
+
+
+def fix(name, df):
+    '''
+    fix the dataframe
+    '''
+
+    # renumber indexes starting at zero
+    df = df.reset_index(drop=True)
+
+    # make an "index" column
+    df = df.reset_index()
+
+    if not (df['Ads_Run'] != 0).any():
+        raise Exception(f'for {name}, we cannot infer a value')
```

Comment:
this brings up a question.
In order for this "fixing" code to work, there has to be at least one "zero" error.
It is not true to say that "if there are no zero errors, the data may not be wrong"

So, there is actually another check that should be made of the data to check for errors.
So this solution is incomplete sadly.

e.g if the sequence (in day order is 5,6,7,8,1,9, etc...  then we know the "1" is wrong even if there are no zero errors.

So in another branch there should be a fix to at least note that this is the case.
and unit tests for it too.